### ISSUE TITLE
Merging `AbstractPlugin` settings gives a plain dict rather than `DottedDict`

### ISSUE THREAD

**Issue by randy3k** [sublimelsp/LSP#1138]
Please include in your report:

* OS and language server

macOS and a custom `AbstractPlugin`

https://github.com/REditorSupport/sublime-ide-r/blob/master/ride/lsp.py#L23

* How you installed LSP (Package Control or from git?)

The branch st-4000-exploration

* Minimal reproduction steps

In `LSP.sublime-settings`, I put something like

```
		"rlang":
		{
			"enabled": true,
			"settings": {
				"debug": true
			}
		}
```

* Log

```
LSP: applying .sublime-project override for rlang
Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python33/sublime_plugin.py", line 220, in unload_module
    module.plugin_unloaded()
  File "/Users/Randy/Dropbox/Sublime Text/Packages/LSP/plugin/core/main.py", line 131, in plugin_unloaded
    unload_sessions(window)  # unloads view state from document sync and diagnostics
  File "/Users/Randy/Dropbox/Sublime Text/Packages/LSP/plugin/core/registry.py", line 68, in unload_sessions
    wm = windows.lookup(window)
  File "/Users/Randy/Dropbox/Sublime Text/Packages/LSP/plugin/core/windows.py", line 645, in lookup
    window_configs = self._configs.for_window(window)
  File "/Users/Randy/Dropbox/Sublime Text/Packages/LSP/plugin/core/configurations.py", line 18, in for_window
    window_configs = WindowConfigManager(window, self._configs)
  File "/Users/Randy/Dropbox/Sublime Text/Packages/LSP/plugin/core/configurations.py", line 33, in __init__
    self.all = self._create_window_configs()
  File "/Users/Randy/Dropbox/Sublime Text/Packages/LSP/plugin/core/configurations.py", line 78, in _create_window_configs
    return [self._apply_project_overrides(c, project_clients_dict) for c in self._global_configs]
  File "/Users/Randy/Dropbox/Sublime Text/Packages/LSP/plugin/core/configurations.py", line 78, in <listcomp>
    return [self._apply_project_overrides(c, project_clients_dict) for c in self._global_configs]
  File "/Users/Randy/Dropbox/Sublime Text/Packages/LSP/plugin/core/configurations.py", line 84, in _apply_project_overrides
    settings = DottedDict(deepcopy(client_config.settings.get()))
```

I did some quick testing to reveal that `client_config.settings` is simple a dict `{"debug": true}` rather than a `DottedDict`. Perhaps when merging the settings from `LSP.sublime-settings` and `AbstractPlugin`, `settings` was not converted to `DottedDict`?

ref: https://github.com/sublimelsp/LSP/blob/f5e38feed10844db51adb7039caf74e9dc3f39b1/plugin/core/settings.py#L285-L285

**Comment by randy3k** [comment]
By the way,

Which of the `LSP.sublime-settings` and `AbstractPlugin ` has higher precedence?

**Comment by rwols** [comment]
Should be fixed by #1140
